packer/linux/conf/buildkite-agent/hooks/environment

```diff
@@ -95,8 +95,13 @@ if [[ "${BUILDKITE_ECR_POLICY:-}" != "none" && "${ECR_PLUGIN_ENABLED:-}" == "1"
     export BUILDKITE_PLUGIN_ECR_ACCOUNT_IDS_0="${AWS_ECR_LOGIN_REGISTRY_IDS}"
   fi
 
+  # Do not propagate the error trap into the ECR plugin's environment hook, because its 'retry' function breaks when we do that.
+  set +E
+
   # shellcheck source=/dev/null
   source /usr/local/buildkite-aws-stack/plugins/ecr/hooks/environment
+
+  set -E
```

Comment:
Set `+E` is a nice fix! But I think you should do this in the ECR environment hook just before calling the `retry` function. Then, the other functions/substitutions/subshells of the ECR plugin will still cause this script to exit with status 53 if they fail, which is something the users that wanted this functionality would desire.